src/include/smf/rpc_server_connection.h

```diff
@@ -71,8 +71,7 @@ class rpc_server_connection : public rpc_connection {
     uint64_t connection_id,
     rpc_server_connection_options opts = rpc_server_connection_options(
       false, false))
-    : rpc_connection(std::move(sock), conn_limits)
-    , remote_address(address)
+    : rpc_connection(std::move(sock), address, conn_limits)
```

Comment:
you can probably do `std::move(address)` for *most* ctor moves. 

so you take it by value - good.
but after you own it in the ctor you can just move it in place.